doc/source/schema/schema.md

```diff
@@ -36,33 +39,73 @@ The terminology used in this specification is based on [xarray's terminology](ht
-* For groups like `observed_data` or `constant_data`, the two initial dimensions(`chain`, `draw`) are omitted.
+* Dimension names `chain`, `draw`, `sample` and `post_pred_id` are reserved for
+  InferenceData use to indicate sample dimensions.
+* Dimensions in InferenceData (including sample dimensions) should be identified by name only.
+* For groups like `observed_data` or `constant_data`, all sample dimensions can be
+  omitted. For groups like `prior`, `posterior` or `posterior_predictive` either `sample` has to be
+  present or both `chain` and `draw` dimensions need to be present. Any combinations that follow
+  this are valid.
 * Dimensions must be named and share name with a coordinate specifying the index values, called coordinate values.
 * Coordinate values can be repeated and should not necessarily be numerical values.
 * Variables must not share names with dimensions.
-* Moreover, each group contains the following attributes:
-  - `created_at`: the date of creation of the group.
-  - `inference_library`: the library used to run the inference.
-  - `inference_library_version`: version of the inference library used.
+* Groups, variables or the InferenceData itself can have arbitrary metadata stored.
 
-### Relations
-`InferenceData` data objects contain any combination of the groups described below. There are also some relations (detailed below) between the variables and dimensions of different groups. Hence, whenever related groups are present they should comply with these relations.
+### Metadata
+No metadata is _required_ to be present in order to be compliant with the InferenceData schema.
+However, it is recommended to store the following fields when relevant:
+* `name`: InferenceData objects represent multiple quantities related to Bayesian modelling,
+  but they are all tied to a single model. The model identifier can be added as metadata
+  to simplify the calls to model comparison functions.
+* `created_at`: the date of creation of the group.
+* `arviz_version`: the version of the ArviZ library that generated the InferenceData
+* `arviz_language`: the programming language from which ArviZ was used to create the InferenceData
+* `inference_library`: the library used to run the inference.
+* `inference_library_version`: version of the inference library used.
+
+
+### Relations between groups
+`InferenceData` data objects contain any combination of the groups described below. There are also some relations (detailed below) between the variables and dimensions of different groups. Hence, whenever related groups are present they should comply with these relations. Neither the presence of groups nor described below or the lack of some of the groups described below go against the schema.
 
 #### `posterior`
-Samples from the posterior distribution p(theta|y).
+Samples from the posterior distribution p(theta|y) in the parameter (also called constrained) space.
```

Comment:
Should this use mathjax?

Reply:
yep, the docs already has mathjax loaded and supports the $ syntax